toffy/mph_comp_test.py

```diff
@@ -0,0 +1,145 @@
+import pandas as pd
+import pytest
+import os
+import tempfile
+import json
+
+from pathlib import Path
+
+from toffy import mph_comp as mph
+
+
+def test_get_estimated_time():
+    bad_path = os.path.join(Path(__file__).parent, "data", "not-a-folder")
+    bad_fov = "not-a-fov"
+
+    good_path = os.path.join(Path(__file__).parent, "data", "tissue")
+    good_fov = 'fov-1-scan-1'
+
+    # bad directory path should raise an error
+    with pytest.raises(ValueError):
+        mph.get_estimated_time(bad_path, good_fov)
+
+    # bad fov name data should raise an error
+    with pytest.raises(FileNotFoundError):
+        mph.get_estimated_time(good_path, bad_fov)
+
+    # bad FOV json file data should raise an error, no frameSize or dwellTimeMillis keys
+    bad_data = {"fov": {"not_frameSize": 0, "not_dwellTimeMillis": 0}}
+    temp_json = tempfile.NamedTemporaryFile(mode="w", suffix='fov_name.json', delete=False)
+    json.dump(bad_data, temp_json)
+    json_dir = tempfile.gettempdir()
+    with pytest.raises(KeyError, match="missing one of the necessary keys"):
+        mph.get_estimated_time(json_dir, 'fov_name')
+
+    # test successful time data retrieval
+    assert mph.get_estimated_time(good_path, good_fov) == 512
+
+
+def test_compute_mph_metrics():
+    bin_file_path = os.path.join(Path(__file__).parent, "data", "tissue")
+    fov_name = 'fov-1-scan-1'
+    target_name = 'CD8'
+    start_mass = -0.3
+    stop_mass = 0.0
+
+    # bad directory path should raise an error
+    bad_path = os.path.join(Path(__file__).parent, "data", "not-a-folder")
+    with pytest.raises(ValueError):
+        mph.compute_mph_metrics(bad_path, bin_file_path, fov_name,
```

Comment:
with all of the `mph.compute_mph_metrics()` calls, you could pass a tempdir into the `csv_dir` argument, so that you don't have to call `os.remove` on all of the created csvs